genie-agent/src/main/java/com/netflix/genie/agent/cli/Util.java

```diff
@@ -63,4 +64,46 @@ private Util() {
     private static String[] replaceAll(final String[] args, final String from, final String to) {
         return Arrays.stream(args).map(arg -> from.equals(arg) ? to : arg).toArray(String[]::new);
     }
+
+    /**
+     * Get a subset of arguments before the double dash (a.k.a. options).
+     *
+     * @param args the raw array of arguments
+     * @return an array of arguments. Possibly empty, possibly the same as the input array.
+     */
+    public static String[] getOptionArguments(final String[] args) {
+        final List<String> temporary = Arrays.asList(args);
+        final int index = temporary.indexOf(BARE_DOUBLE_DASH);
```

Comment:
It's late and I'm tired so I could be totally off on this but I feel like this could be done in a single pass and save all the conversions and checks. Maybe just iterate through the array and write each item into a resultant array until you encounter the double dash and then return the result array?

Reply:
This could indeed be a lot simpler code at the cost of allocating a couple more objects.
Instinctively went for the more memory efficient, but probably not worth in retrospect, since it's only ever invoked once.

Reply:
Simplified